sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -219,29 +242,55 @@ def incr_mean_variance_axis0(X, last_mean, last_var, unsigned long last_n):
     `utils.extmath._batch_mean_variance_update`.
 
     """
-    cdef unsigned long n_samples = X.shape[0]
-    cdef unsigned int n_features = X.shape[1]
+    if X.dtype == np.int32 or X.dtype == np.int64:
+        X = X.astype(np.float64)
+    return _incr_mean_variance_axis0(X.data, X.shape, X.indices, X.indptr,
+                                     last_mean, last_var, last_n)
+
+
+def _incr_mean_variance_axis0(np.ndarray[floating, ndim=1] X_data,
+                              shape,
+                              np.ndarray[int, ndim=1] X_indices,
+                              np.ndarray[int, ndim=1] X_indptr,
+                              last_mean, last_var, unsigned long last_n):
```

Comment:
So what type should we use here for `last_mean` and `last_var`, `np.ndarray[np.float64, ndim=1]` ?

In the [test](https://github.com/scikit-learn/scikit-learn/pull/6593/files#diff-274f1031827d26cb7a055b88042b45f1L101), these two arguments are always `np.ndarray[np.float64, ndim=1]`